flake8 complexity check and ammend validation logic on reigster and login.

Eric-Butcher opened the pull request:
…t does not flag in-built django files.

Eric-Butcher committed:
Added flake7 function complexity checking to most strict level that it does not flag in-built django files.

WeirdAlex03 commented on evergreen/.flake8:
```diff
@@ -2,3 +2,4 @@
 max-line-length = 88
 extend-ignore = E203,E701
 exclude = .git,__pycache__,.venv,venv
```
You can exclude `./evergreen/staticfiles/admin` if you want a higher standard for us without annoyances from Django

Eric-Butcher replied:
done

WeirdAlex03 committed:
Rework Docker & nginx configs for w/ & w/o SSL (#53)

sumanyuj committed:
Project Part 2: Web Sockets: WEB SOCKET SUPREMACY (#50)

* adding channels to poetry

* added middleware and layer to settings.py

* created consumers.py to handle web sockets

* routing for web sockets

* updated asgi to handle web sockets in django

* working websockets

* minor fixes

* converted like button to websockets

* run make some fixes

* fixed exception handler

* removed unused imports

* ran make

* changed

* fixed black

* commented out cmd

* removed comment

* Update evergreen/eg_app/consumers.py

Co-authored-by: Alex M <weirdalex0302+github@gmail.com>

* fixed issue with web sockets

* ran make

* Update urls.py

* Update functions.js

* removed allow all orgins

* ran make

* removed duplicate like button

* updated poetry lock

* increase static file limit nginx

* passing in the self signed cert files

* updated front end logic and visual for updated like button and uploading content via sockets

* updated  backend to handle websockets handling content, like button logic, and more!

* running make

* poetry lock fix

* added toml

* Update evergreen/staticfiles/js/functions.js

Co-authored-by: Alex M <weirdalex0302+github@gmail.com>

* removed blank lines

* fixed exception

* minor image access check

* better file type checking, making sure to actually check if image is uplaoded, magic numbers, and a bit more!

* cleanup logs

* removed comment

* fixed pesky braket issue (red alert)

* rules of engagement

* ran make

* go rid of styles

* quick fix to limit text size and image size on backend (and alert user)

* f-strings baby

* ran magic make

* removed ssl stuff and updated rules of gentlemanly plant engagement

* removed 8MB limit that was non working

* type annoation

* restored nginx ignore

---------

Co-authored-by: Alex M <weirdalex0302+github@gmail.com>

sumanyuj committed:
fixed escaping (#54)

WeirdAlex03 committed:
Nuke @csrf_exempt: fix problem instead of symptom (#56)

Just had to add "http://localhost" to Django CSRF_TRUSTED_ORIGINS.
That's it.

giagermain committed:
Hotfix/redirect (#57)

* fixed escaping

* added redirects note and url to read me

* i cant spell

---------

Co-authored-by: Sumanyu Janapareddy <111717258+sumanyuj@users.noreply.github.com>

WeirdAlex03 committed:
Add media folder to gitignore (#58)

Honestly idk why we didn't have this already

sumanyuj committed:
updated readme for env (#59)

* updated readme for env

* Update README.md

sumanyuj committed:
Demo Fix: Catch Socket Frames Before They Inflict Maximum Damage (#61)

* catches frame in recieve method to check for if frame size is exceeded or not.

* added csrf exempt to logout view!!

* ran make

giagermain committed:
fixed nontransparent background (#64)

* fixed nontransparent background

* added redirects

WeirdAlex03 committed:
Rate limiting for demo (#60)

* tmp basic njs starter pending piazza ans

* nginx rate limiting for demo

this is totally incompatible with the AO,
but should hopefully help keep us online in the demo
:fingers_crossed: :fingers_crossed: :fingers_crossed:

* demo ip

Eric-Butcher committed:
Feature/camera relay (#62)

* Added ignores so my IDE doesn't yell at me.

* Creared relay_handler model.

* Fixed relay_request to use the request.POST variable instead.

* Added command to create new RelayDevic.

* Returns an error message if the relay handler failed to add timelapse to psotgres.

* Fixed exception handling in relay_processor.py.

* Reigster models in admin file.

* URL for the timelapse.

* Fixing linter and type-checking erroring.

* Time lapse now has lcoation on the main page.

* Added relay upload path.

* Made some fixes

* Fixed large files being 413'd.

* Linting with make to fix bugs and error codes from flake8 about unused imports. Also added max_length to models.

giagermain committed:
Hotfix/redirect (#65)

* fixed escaping

* added redirects note and url to read me

* i cant spell

* fixed readme

* small typo

* fixed clunkiness

---------

Co-authored-by: Sumanyu Janapareddy <111717258+sumanyuj@users.noreply.github.com>

Eric-Butcher committed:
Feature/camera relay (#66)

Added README testing procedure for Part3

WeirdAlex03 committed:
Drop breaking line from nginx config (#67)

Weren't using that anyway, just a leftover from some prior stuff

Eric-Butcher committed:
Narrowed try/except context

Eric-Butcher committed:
fixing login and registering functions complexity

Eric-Butcher committed:
Unit Testing for new validators functionality.'

Eric-Butcher committed:
Updated .flake8 to use a cyclomatic complexity equal to 10, as recommended by this wiki article on the creator and topic.

WeirdAlex03 commented on evergreen/eg_app/util/validators.py:
```diff
@@ -50,3 +52,23 @@ def validate_password(password: str) -> bool:
+
+
+def validate_password_pair(password: str, password_confirmation: str):
+    return hmac.compare_digest(password, password_confirmation) and validate_password(
+        password
+    )
```
What is the use of `hmac.compare_digest()` over plain `==`?

If you need it, I'd break the line as
```suggestion
    return hmac.compare_digest(password, password_confirmation)
        and validate_password(password)
```

Also, all these changes have nothing to do with the listed purpose of the PR. They're good changes, but you should rename the PR (or break these out into a new PR entirely if you to be really fancy)

Eric-Butcher replied:
Yea now that I look at this again I don't think it is necessary. Just a habit of using hmac when doing authentication.

Eric-Butcher replied:
That is just the way black formats it and I don't believe your change is worth the hassle of adding a noqa even though it is prettier that way you suggested.

WeirdAlex03 replied:
I assume it just tries to break the line as close to the limit as possible, which in this case meant separating the argument. I hope now that both lines are good, it shouldn't try to merge them together and make that mess again, right?

Either way, if hmac isnt needed then its a moot point, with `==` it should all fit

Eric-Butcher replied:
I changed from hmac.compare_digest() to ==.

WeirdAlex03 commented on evergreen/eg_app/views.py:
```diff
@@ -108,62 +108,45 @@ def addCookies(response: HttpResponse, cookies: dict[str, str]) -> HttpResponse:
-
-        if not val.validate_password(password):
-            valid_pass = False
+@csrf_exempt
```
I don't want to be adding more `@csrf_exempt`s at this point now that we aren't worrying about deadlines. We need to address the root cause instead of blindly throwing more bandaids.

Eric-Butcher replied:
Was already there before me, this is git just tracking things oddly. It is necessary because a person cannot have their csrf token if they have not even registered yet. This is the problem of having the same main page be the signin/up.

WeirdAlex03 replied:
In that case we should probably make separate signup/login pages. Django has built-in tools to make that fairly simple

https://learndjango.com/tutorials/django-login-and-logout-tutorial#log-in-page
https://www.pythontutorial.net/django-tutorial/django-loginview/

(Though both of these also use csrf tokens, so I'm not sure that's actually the problem?)

Eric-Butcher replied:
We can do this later.

Eric-Butcher replied:
Never mind I think that somehow snuck in when I merged and didn't see it.

WeirdAlex03 commented on evergreen/eg_app/tests/util/test_validators.py:
```diff
@@ -58,3 +70,156 @@ def test_valid_passwords(self):
         self.assertTrue(validate_password("a" * 12))
         self.assertTrue(validate_password("1" * 12))
         self.assertTrue(validate_password("A" * 255))
+        self.assertTrue(string.ascii_lowercase, string.ascii_lowercase)
+        self.assertTrue(string.ascii_letters, string.ascii_letters)
```
What are these two testing?

Eric-Butcher replied:
I guess the first one isn't really necessary. The string class just comes with a nice set of collections of common strings so I was making sure that all ascii input works.

WeirdAlex03 replied:
(neither are actually testing our methods, you're just comparing each to itself. and not even that, b/c its assertTrue so youre just checking the truthiness of a string and printing that string if it fails)

Eric-Butcher replied:
fixed

WeirdAlex03 commented on evergreen/eg_app/tests/util/test_validators.py:
```diff
@@ -24,6 +33,7 @@ def test_valid_ubits(self):
         self.assertTrue(validate_email("alfanum1@buffalo.edu"))
         self.assertTrue(validate_email("lt12@buffalo.edu"))
         self.assertTrue(validate_email("hello123@buffalo.edu"))
+        self.assertTrue(validate_email("doermann@buffalo.edu"))
```
I intentionally didn't use real emails. He isn't associated with this at all and shouldn't be included in the source code

Eric-Butcher replied:
This is a fair point.

Eric-Butcher replied:
fixed

WeirdAlex03 commented on evergreen/eg_app/tests/util/test_validators.py:
```diff
@@ -58,3 +70,156 @@ def test_valid_passwords(self):
+        ubit = User.objects.create_user(
+            username=ubit_email, email=ubit_email, password="thisissecure"
+        )
+        ubit.save()
```
Same as above re: real staff emails

Eric-Butcher replied:
fixed

WeirdAlex03 reviewed (changes requested):
Generally good addition, but a few notes on content and scope

Eric-Butcher committed:
Merge branch 'dev' into hotfix/flake8-complexity-check

Eric-Butcher committed:
Fixed linter error.

Eric-Butcher commented:
I did an interactive rebase to fix the testing names and force pushed the changes. I will have to retest now. The goal of this branch is really just to get the repo into a good enough state with everything cleaned up so we can finally PR to main before we archive this repository on github and start doing continuing development for the club on a new repo on the gh organization.

Eric-Butcher committed:
Changed to == from hmac.compare_digest() as requested. The digest was not needed since this section of the code should not be subject to a timing attack.

Eric-Butcher committed:
Removed unused hmac import.
